definitions/tools/fastq_align_and_tag.cwl

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "align with bwa_mem and tag"
+
+baseCommand: ["/bin/bash", "fastq_alignment_helper.sh"]
+requirements:
+    - class: ResourceRequirement
+      coresMin: 8
+      ramMin: 20000
+    - class: DockerRequirement
+      dockerPull: "mgibio/alignment_helper-cwl:1.0.0"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'fastq_alignment_helper.sh'
+        entry: |
+            set -o pipefail
+            set -o errexit
+            set -o nounset
+
+            FASTQ1="$1"
+            FASTQ2="$2"
+            READGROUP="$3"
+            REFERENCE="$4"
+            NTHREADS="$5"
+
+            /usr/local/bin/bwa mem -K 100000000 -t "$NTHREADS" -Y -p -R "$READGROUP" "$REFERENCE" "$FASTQ1" "$FASTQ2" | /usr/local/bin/samblaster -a --addMateTags | /opt/samtools/bin/samtools view -b -S /dev/stdin
+stdout: "refAlign.bam"
+arguments:
+    - position: 5
+      valueFrom: $(runtime.cores)
+inputs:
+    fastq_1:
+        type: File
+        inputBinding:
+            position: 1
+        doc: 'fastq of sequence data (first read if paired end)'
+    fastq_2:
```

Comment:
Would having this as optional throw off the positions of the input variables for the bash script? and actually cause the bash script to fail because an input variable doesn't exist?

Reply:
Yeah--you're right; the current script doesn't work right without a second FASTQ.  I'll just make it required for now; we can revisit if someone really wants to process fragment data.